docs/backup-restore.md

```diff
@@ -0,0 +1,109 @@
+# Backup and Restore TiDB Cluster
+
+## Overview
```

Comment:
Please remove this unnecessary line. An intro para that directly follows the title is a rule for a standard tech document.